src/sql/parts/notebook/cellViews/codeActions.ts

```diff
@@ -26,4 +29,50 @@ export class RunCellAction extends Action {
 			}
 		});
 	}
+}
+
+export class AddCellAction extends Action {
+	constructor(
+		id: string, label: string, private cellType: CellType, private isAfter: boolean
+	) {
+		super(id, label);
+	}
+	public run(model: NotebookModel): TPromise<boolean> {
+		return new TPromise<boolean>((resolve, reject) => {
+			try {
+				if (!model) {
+					return;
+				}
+				let index = model.cells.findIndex((cell) => cell.id === model.activeCell.id);
+				if (index !== undefined && this.isAfter) {
+					index += 1;
+				}
+				model.addCell(this.cellType, index);
+			} catch (error) {
+				let message = getErrorMessage(error);
+				//ApiWrapper.showErrorMessage(message);
```

Comment:
You will need to inject the `@INotificationService` like we do elsewhere. Then please uncomment and show the error.